test/e2e/node/mount_propagation.go

```diff
@@ -191,9 +191,13 @@ var _ = SIGDescribe("Mount propagation", func() {
 		cmd = "pidof kubelet"
 		kubeletPid, err := hostExec.IssueCommandWithResult(ctx, cmd, node)
 		framework.ExpectNoError(err, "Checking kubelet pid")
-		kubeletPid = strings.TrimSuffix(kubeletPid, "\n")
-		gomega.Expect(strings.Count(kubeletPid, " ")).To(gomega.Equal(0), "kubelet should only have a single PID in the system (pidof returned %q)", kubeletPid)
-		enterKubeletMountNS := fmt.Sprintf("nsenter -t %s -m", kubeletPid)
+		ginkgo.By(fmt.Sprintf("pidof kubelet returns %s", kubeletPid))
+		// In rare cases in the CI, we are seeing two pids for kubelet.
+		// We did some investigating and the last entry of this list is the one that we want
+		// Higher pids are more likely to be created after the first one
+		kubeletPids := strings.Split(kubeletPid, " ")
+		enterKubeletMountNS := fmt.Sprintf("nsenter -t %s -m", kubeletPids[len(kubeletPids)-1])
+		ginkgo.By(enterKubeletMountNS)
```

Comment:
I don't think it is the best use of a `ginkgo.By`. It is typically used to describe test steps. Perhaps `framework.Logf` can be used instead?

Reply:
updated.